MediaBrowser.Controller/LibraryTaskScheduler/LimitedConcurrencyLibraryScheduler.cs

```diff
@@ -96,14 +96,33 @@ async Task RunCleanupTask()
         }
     }
 
+    private bool ShouldForceSequencialOperation()
+    {
+        // if the user either set the setting to 1 or its unset and we we have less then 4 cores its better to run sequencially.
+        var fanoutSetting = _serverConfigurationManager.Configuration.LibraryScanFanoutConcurrency;
+        return fanoutSetting is 1 || (fanoutSetting <= 0 && Environment.ProcessorCount <= 3);
+    }
+
+    private int CalculateScanConcurrencyLimit()
+    {
+        // when this is invoked, we already checked ShouldForceSequencialOperation for the sequencial check.
+        var fanoutConcurrency = _serverConfigurationManager.Configuration.LibraryScanFanoutConcurrency;
+        if (fanoutConcurrency <= 0)
+        {
+            // in case the user did not set a limit manually, we can assume he has 3 or more cores as already checked by ShouldForceSequencialOperation.
```

Comment:
```suggestion
            // in case the user did not set a limit manually, we can assume he has 3 or more cores as already checked by ShouldForceSequentialOperation.
```